src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs

```diff
@@ -594,34 +587,28 @@ protected virtual void OnCustomCommand(int command)
         ///       contains multiple associated services. Loads the specified services into memory so they can be
         ///       started.</para>
         /// </devdoc>
-        public static void Run(ServiceBase[] services)
+        public static unsafe void Run(ServiceBase[] services)
         {
             if (services == null || services.Length == 0)
                 throw new ArgumentException(SR.NoServices);
 
-            int sizeOfSERVICE_TABLE_ENTRY = Marshal.SizeOf<SERVICE_TABLE_ENTRY>();
-
-            IntPtr entriesPointer = Marshal.AllocHGlobal(checked((services.Length + 1) * sizeOfSERVICE_TABLE_ENTRY));
+            IntPtr entriesPointer = Marshal.AllocHGlobal(checked((services.Length + 1) * Marshal.SizeOf<SERVICE_TABLE_ENTRY>()));
+            Span<SERVICE_TABLE_ENTRY> entries = new Span<SERVICE_TABLE_ENTRY>((void*)entriesPointer, services.Length + 1);
             try
             {
-                SERVICE_TABLE_ENTRY[] entries = new SERVICE_TABLE_ENTRY[services.Length];
                 bool multipleServices = services.Length > 1;
-                IntPtr structPtr;
 
-                for (int index = 0; index < services.Length; ++index)
+                for (int index = 0; index < entries.Length - 1; ++index)
```

Comment:
Nit: I think it would read better with services:
```suggestion
                for (int index = 0; index < services.Length; ++index)
```